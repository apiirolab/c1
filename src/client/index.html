<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>