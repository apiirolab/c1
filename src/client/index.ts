export const info = 'Client Info';
