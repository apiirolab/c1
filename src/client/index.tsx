import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
