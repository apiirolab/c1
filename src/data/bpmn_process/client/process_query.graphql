query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
