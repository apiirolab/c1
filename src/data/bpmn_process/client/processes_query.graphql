query Processes {
  processes {
    id
    name
  }
}
