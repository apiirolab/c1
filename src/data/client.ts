import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
