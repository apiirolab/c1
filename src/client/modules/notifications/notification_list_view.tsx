import * as React from 'react';

import { inject, observer } from 'mobx-react';
import { CSSTransitionGroup } from 'react-transition-group';
import { Button, Header, List, Message } from 'semantic-ui-react';

import { QueryTypes } from 'data/client';
import { NotificationView } from './notification_item_view';
import { RemoveNotificationsMutation } from './notifications_queries';

type Props = {
  notifications: QueryTypes.NotificationsQuery_Notifications[];
  context?: App.Context;
};

let notification: QueryTypes.NotificationsQuery_Notifications;

export const NotificationsListView: React.SFC<Props> = ({ notifications, context }) => (
  <>
    <Choose>
      <When condition={notifications && notifications.length > 0}>
        <>
          <Header content="Notifications" icon="bell" as="h5" dividing />
          <List relaxed divided>
            <CSSTransitionGroup
              component={React.Fragment}
              transitionName="example"
              transitionAppear={true}
              transitionAppearTimeout={500}
              transitionEnter={false}
              transitionEnterTimeout={500}
              transitionLeave={true}
              transitionLeaveTimeout={500}
            >
              <For each="notification" of={notifications}>
                <NotificationView key={notification.id} notification={notification} />
              </For>
            </CSSTransitionGroup>
          </List>
          <RemoveNotificationsMutation>
            {clearNotifications => (
              <Button
                onClick={() => clearNotifications()}
                color="red"
                icon="trash"
                labelPosition="left"
                content={context.i18n`Clear All`}
              />
            )}
          </RemoveNotificationsMutation>
        </>
      </When>
      <Otherwise>
        <Message>You have no new notifications.</Message>
      </Otherwise>
    </Choose>
  </>
);

NotificationsListView.displayName = 'NotificationsListView';

export const NotificationsListContainer = inject('context')(observer(NotificationsListView));
