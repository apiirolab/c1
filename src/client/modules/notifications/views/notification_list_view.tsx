import * as React from 'react';
import { Header, List, ListProps, Message, Segment } from 'semantic-ui-react';
import styled, { StyledComponentClass } from 'styled-components';

import { Yoga } from 'data/yoga';
import { NotificationView } from './notification_item_view';

const NotificationList: StyledComponentClass<ListProps, {}> = styled(List)`
  &&&&& .item {
    line-height: inherit;
  }
`;

type Props = {
  notifications: Yoga.Notification[];
};

let notification: Yoga.Notification;

export const NotificationListView: React.SFC<Props> = ({ notifications }) => (
  <React.Fragment>
    <Choose>
      <When condition={notifications && notifications.length > 0}>
        <Segment>
          <Header content="Notifications" icon="bell" as="h5" />
          <NotificationList divided>
            <For each="notification" of={notifications}>
              <NotificationView key={notification.id} notification={notification} />
            </For>
          </NotificationList>
        </Segment>
      </When>
      <Otherwise>
        <Message>You have no new notifications.</Message>
      </Otherwise>
    </Choose>
  </React.Fragment>
);
