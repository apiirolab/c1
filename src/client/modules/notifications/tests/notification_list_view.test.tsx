import * as React from 'react';

import { create as render } from 'react-test-renderer';

import { create, MockedProvider, QueryTypes } from 'client/tests';
import { NotificationsListContainer } from '../notification_list_view';
import { createNotification } from './notifications_test_data';

export const notificationData = [
  createNotification({ id: '1', createdAt: new Date(2010, 1, 2, 12, 50) }),
  createNotification({ id: '2', createdAt: new Date(2010, 1, 2, 3, 50) }),
  createNotification({ id: '3', createdAt: new Date(2007, 1, 2, 10, 50) }),
  createNotification({ id: '4', createdAt: new Date(2010, 1, 2, 11, 50) }),
  createNotification({ id: '5', createdAt: new Date(2011, 1, 2, 12, 50) })
];

describe('Notifications', () => {
  describe('List', () => {
    function componentWithData(notifications: QueryTypes.Notifications[]) {
      return (
        <MockedProvider>
          <div>
            <NotificationsListContainer notifications={notifications} context={create.context()} />
          </div>
        </MockedProvider>
      );
    }

    describe('Empty', () => {
      it('renders no notifications', () => {
        const root = render(componentWithData([]));
        expect(root).toMatchSnapshot();
      });

      return {
        componentWithData
      };
    });

    describe('Data', () => {
      it('renders some notifications', () => {
        const root = render(componentWithData(notificationData));
        expect(root).toMatchSnapshot();
      });

      return {
        component: componentWithData(notificationData)
      };
    });
  });
});
