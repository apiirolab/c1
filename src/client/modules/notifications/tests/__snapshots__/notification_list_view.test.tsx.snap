// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Item View renders no notifications 1`] = `
<div
  className="ui message"
>
  You have no new notifications.
</div>
`;
